### ISSUE TITLE
PhoneInput EndEnhancer not styleable in most recent Baseweb

### ISSUE THREAD

**Issue by houmark** [uber/baseweb#3713]
## Current Behavior

In earlier Baseweb versions it was possible to add an EndEnhancer to `PhoneInput`, though it seems this was either not documented or part of the "playground" for that component.

Nonetheless it worked and we have some buttons on the right of some PhoneInputs.

In the most recent version it no longer seems possible to style the EndEnhancer in any way using overrides. For example a background change is required in our UI for the buttons to render nicely.

We also noticed changes to other fields for EndEnhancers, where the background is no longer black (design change in Baseweb?) but as a minor update that breaks the visuals of the design. Also the changelog seems to be a few releases behind so it's even harder to spot these potentially breaking changes.

| Tech    | Version |
| ------- | ------- |
| Base UI | v9.98.0  |
| React   |    16.13.1     |
| browser |    Any     |

- [x] I have searched the [issues](https://github.com/uber/baseweb/issues) of this repository and believe that this is not a duplicate.

**Comment by houmark** [comment]
After testing a bit more I can confirm that this changed between 9.97.0 and 9.98.0, most likely in commit ee54ebf8ca30cc63bbe8fc21e9b1120d9321802a.

For some fields, I've been able to apply the following override with success, but it doesn't work for `PhoneInput`:

```
overrides={{ Root: { style: { paddingRight: 0 } }, EndEnhancer: { style: { backgroundColor: '#000', color: '#fff' } } }}
```

Also note that a padding was added, which makes buttons and icons with backgrounds not good looking visually as the input adds a 14px padding in the right of the EndEnhancer.

**Comment by houmark** [comment]
FWIW, I was able to override the EndEnhancer using a nested override on the `PhoneInput`:

```
overrides={{
          Input: {
            props: {
              overrides: {
                Root: { style: { paddingRight: 0 } },
                EndEnhancer: {
                  style: { backgroundColor: '#000', color: '#fff' },
                },
              },
            },
          },
        }}
```

**Comment by jiaoy14** [comment]
Yeah, thanks for pointing this out. We are currently making design changes to several components. As you can see, we make some spacing and color change. 

We are sorry for the trouble for those who applied the previous design. Nested override is a good way for these.

**Comment by sandgraham** [comment]
@houmark could you add some visuals and a repro to better see what the issue is? Otherwise it seems like you've found an override path to get this working and I'll close this out.
